In [1]:
print('Takumi House & Kailen Santos')

Takumi House & Kailen Santos


In [2]:
from matplotlib import pyplot as plt
import matplotlib as mp
import numpy as np
import pandas as pd
from sklearn.cluster import KMeans
import sklearn.manifold
from scipy import cluster
import seaborn as sns
import sklearn.datasets as sk_data
import sklearn.metrics as metrics
import requests
import json
import nltk.classify.util
from nltk.classify import NaiveBayesClassifier
from nltk.corpus import stopwords
from nltk.stem.porter import PorterStemmer
from nltk import word_tokenize
import re
import json
import pickle
%matplotlib inline

In [3]:
try:
    with open('df_reviews','rb') as f:
        df_reviews = pickle.load(f)
        
except FileNotFoundError:
    with open('df_reviews','wb') as f:
        pickle.dump(df_reviews, f)

In [4]:
try:
    with open('stemmed_data','rb') as f:
        stemmed_data = pickle.load(f)
        
except FileNotFoundError:
    stemmed_data = [" ".join(SnowballStemmer("english", ignore_stopwords=True).stem(word)  
        for sent in sent_tokenize(message)
        for word in word_tokenize(sent))
        for message in df_reviews['text']]
    
    with open('stemmed_data','wb') as f:
        pickle.dump(stemmed_data, f)

In [5]:
df = pd.read_csv('Restaurant_Inspections.csv',header=0)
# df = df[df.City == 'Las Vegas']
# df = (df[df['Inspection Result'] == 'Closed with Fees']).append(df[df['Inspection Result'] == 'Closed without Fees'])

df.shape

#with open ('Resaurant_Inspections.csv') as file:
#   for line in file:
#      if (line[6]=='Las Vegas' && line[]

C:\Users\Kailen\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2717: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


(123919, 23)

In [6]:
df = df[df.City == 'Las Vegas']
df.shape

(94700, 23)

In [7]:
categories = df['Category Name'].unique().tolist()
print(categories)

['Pantry', 'Special Kitchen', 'Garde Manger', 'Kitchen Bakery', 'Bar / Tavern', 'Buffet', 'Snack Bar', 'Restaurant', 'Banquet Support', 'Portable Unit', 'Banquet Kitchen', 'Portable Bar', 'Meat/Poultry/Seafood', 'Vegetable Prep', 'Caterer', 'Barbeque', 'Confection', 'Concessions', 'Produce Market', 'Institutional Food Service', 'Elementary School Kitchen', 'Self-Service Food Truck', 'Bakery Sales', 'Food Trucks / Mobile Vendor', 'Grocery Store Sampling', 'Beer Bar', 'Childcare Kitchens', "Farmer's Market", 'Main Kitchen', 'Frozen Meat Sales']


In [8]:
categories.remove('Elementary School Kitchen')
categories.remove('Grocery Store Sampling')
categories.remove('Farmer\'s Market')
categories.remove('Caterer')
categories.remove('Concessions')
categories.remove('Institutional Food Service')
categories.remove('Banquet Support')

In [9]:
df = df[df['Category Name'].isin(categories)]
df.shape

(91735, 23)

In [10]:
coord_strings = df['Location 1'].tolist()

In [11]:
import ast
xcoords = []
ycoords = []
for coord in coord_strings:
    xcoords.append(ast.literal_eval(coord)[0])
    ycoords.append(ast.literal_eval(coord)[1])

In [12]:
count = 0
for idx, ycoord in enumerate(ycoords):
    if ycoord > 0:
        ycoords[idx] = ycoord * -1
        count += 1
print('{} coordinates fixed.'.format(count))

574 coordinates fixed.


In [13]:
df['Coordinates'] = list(zip(xcoords, ycoords))
df.head()

Serial Number Permit Number                 Restaurant Name  \
11     DA0001715     PR0000426  CAESARS AUGUSTUS BUTLER PANTRY   
12     DA0001716     PR0000426  CAESARS AUGUSTUS BUTLER PANTRY   
13     DA0001728     PR0000427  CAESARS DIAMOND LOUNGE KITCHEN   
14     DA0001773     PR0000434         CAESARS BAKERY WAREWASH   
15     DA0001776     PR0000434         CAESARS BAKERY WAREWASH   

                    Location Name    Category Name                Address  \
11  CAESARS PALACE HOTEL & CASINO           Pantry  3570 S Las Vegas Blvd   
12  CAESARS PALACE HOTEL & CASINO           Pantry  3570 S Las Vegas Blvd   
13  CAESARS PALACE HOTEL & CASINO  Special Kitchen  3570 S Las Vegas Blvd   
14  CAESARS PALACE HOTEL & CASINO  Special Kitchen  3570 S Las Vegas Blvd   
15  CAESARS PALACE HOTEL & CASINO  Special Kitchen  3570 S Las Vegas Blvd   

         City   State         Zip  Current Demerits  \
11  Las Vegas  Nevada  89109-8924                 3   
12  Las Vegas  Nevada  89109-8924                 3   
13  Las Vegas  Nevada  89109-8924                 8   
14  Las Vegas  Nevada  89109-8924                 3   
15  Las Vegas  Nevada  89109-8924                 3   

               ...             Employee ID     Inspection Type  \
11             ...               EE7000417  Routine Inspection   
12             ...               EE7000567  Routine Inspection   
13             ...               EE7000417  Routine Inspection   
14             ...               EE7000417  Routine Inspection   
15             ...               EE7000392  Routine Inspection   

   Inspection Demerits Inspection Grade Permit Status Inspection Result  \
11                   0                A           NaN         Compliant   
12                   2                A           NaN         Compliant   
13                   4                A           NaN         Compliant   
14                   7                A           NaN         Compliant   
15                   2                A           NaN         Compliant   

    Violations          Record Updated                  Location 1  \
11         NaN  02/21/2013 10:26:12 PM  (36.1161559, -115.1750576)   
12         NaN  02/21/2013 10:26:12 PM  (36.1161559, -115.1750576)   
13          14  02/21/2013 10:26:12 PM  (36.1161559, -115.1750576)   
14    14232831  01/22/2014 02:03:44 PM  (36.1161559, -115.1750576)   
15         NaN  01/22/2014 02:03:44 PM  (36.1161559, -115.1750576)   

                   Coordinates  
11  (36.1161559, -115.1750576)  
12  (36.1161559, -115.1750576)  
13  (36.1161559, -115.1750576)  
14  (36.1161559, -115.1750576)  
15  (36.1161559, -115.1750576)  

[5 rows x 24 columns]

In [14]:
coordinates_list = df['Coordinates'].tolist()
xcoords = []
ycoords = []
for coordinates in coordinates_list:
    xcoords.append(coordinates[0])
    ycoords.append(coordinates[1])
df['X'] = xcoords
df['Y'] = ycoords

In [15]:
df = df[df['Y'] < -114.5]
df = df[df['Y'] > -115.6]
df_clean = df.reset_index(drop = True)

In [16]:
coordinates_list = df_clean['Coordinates'].tolist()
xcoords = []
ycoords = []
for coordinates in coordinates_list:
    xcoords.append(coordinates[0])
    ycoords.append(coordinates[1])

In [17]:
df = df_clean[df_clean['Date Current']==df_clean['Inspection Date']]
df

Serial Number Permit Number                            Restaurant Name  \
63        DA0003413     PR0000492   CAESARS SEASONAL VENUS POOL BAR  DELETED   
196       DA0009343     PR0000728  MIRAGE 1 OAK SMALL BAR 2 ORIGINAL DELETED   
254       DA0011310     PR0000730     MIRAGE 1 OAK MULTILEVEL BAR 4  DELETED   
317       DA0017310     PR0000631                  Golden Gate Beverage Cart   
348       DA001XLIF     PR0022088                         Marketon #1 - Deli   
436       DA0023482     PR0001171                         Aramark Nacho Cart   
503       DA0025038     PR0001066             Kentucky Fried Chicken #527621   
505       DA0025516     PR0001062         Kentucky Fried Chicken / Pizza Hut   
531       DA0027055     PR0001064            Kentucky Fried Chicken #X527030   
533       DA0027285     PR0001068             Kentucky Fried Chicken #527019   
557       DA0029175     PR0001074             Kentucky Fried Chicken #527024   
558       DA0029576     PR0001659                     K Mart #4369 Snack Bar   
592       DA0039728     PR0001891                CIRCUS MEXITALIA XPRESS BAR   
800       DA0055718     PR0000251                           Java in the Park   
891       DA0059162     PR0000322                Circle K Store #1248 - Deli   
1100      DA0093418     PR0002881                       VONS #1963 STARBUCKS   
1103      DA0093785     PR0002411                    Marie Callender's's #72   
1104      DA0093832     PR0002412             Marie Callender's's #72 Saloon   
1105      DA0093835     PR0002413      Marie Callender's's #72 Sunday Brunch   
1219      DA00E4NAR     PR0014994          OCEAN ONE BAR & GRILLE RESTAURANT   
1220      DA00KFO00     PR0104466                       WET N WILD SNACK BAR   
1221      DA00VEPXC     PR0006003                    HARD ROCK SANS POOL BAR   
1288      DA0104852     PR0001994                       White Cross Drugs #2   
1437      DA0113862     PR0002678                                Dairy Queen   
1493      DA0120049     PR0002749                  Carl's Jr Restaurant #883   
1551      DA0123453     PR0002800                      L a Express Ice Cream   
1743      DA0134595     PR0003488                          Burger King #2701   
1842      DA0138293     PR0003292                     7-Eleven #26637 - Deli   
1867      DA0142409     PR0003124                            Rascals Kitchen   
2014      DA0185680     PR0004240                                     Champs   
...             ...           ...                                        ...   
89526     DARL3H49O     PR0121917                       ANTOJITOS D.F. Y MAS   
89594     DAUDJGYGL     PR0119548                     Dotty's #138 SNACK BAR   
89650     DAVCPCHBE     PR0121950   HARD ROCK MAGIC MIKE'S LIVE BASEMENT BAR   
89662     DAVQQPOKX     PR0106985                 OFF THE STRIP PREP STORAGE   
89664     DAVSYBOOX     PR0120107      SUGAR FACTORY - UPSTAIRS DAIQUIRI BAR   
89667     DAVUOAFLR     PR0119884                            JACK IN THE BOX   
89669     DAVVC4M8B     PR0122607                WARES CANDY ICE CREAM TRUCK   
89685     DAWIIAJ0G     PR0122379     PLAZA HOTEL BRIGHTSIDE CAFE - COOKLINE   
89707     DAX7GAFWI     PR0121685                               INSPIRE PUSF   
89723     DAVE6DSQV     PR0121651                              FOOD TO HOMES   
89942     DAORFUG7L     PR0116193                   NUTRITION RUSH SNACK BAR   
90052     DAN46ZW1B     PR0122301                LEMON HEAVEN @ LVPO #2 PUSF   
90229     DALQAH6FS     PR0119843                              DISTILL - BAR   
90295     DAPSS2J5L     PR0120208          LA BONITA SUPERMARKETS RESTAURANT   
90389     DAA9MDFFM     PR0121622                               TAIL AND FIN   
90628     DA1763489     PR0116428                  MODERNA TEA @ FORUM SHOPS   
90672     DANI6SDQJ     PR0121470                        SUBWAY STORE #53283   
90686     DAXSEOVM5     PR0120738                          JABURRITOS @ LINQ   
90690 

In [18]:
df_clean = df.reset_index(drop=True)

In [19]:
df_agg = df_clean.groupby('Restaurant Name',as_index=False).agg(lambda x: x.tolist())

In [20]:
dfcategories = df_agg.columns
dfcategories

Index(['Restaurant Name', 'Serial Number', 'Permit Number', 'Location Name',
       'Category Name', 'Address', 'City', 'State', 'Zip', 'Current Demerits',
       'Current Grade', 'Date Current', 'Inspection Date', 'Inspection Time',
       'Employee ID', 'Inspection Type', 'Inspection Demerits',
       'Inspection Grade', 'Permit Status', 'Inspection Result', 'Violations',
       'Record Updated', 'Location 1', 'Coordinates', 'X', 'Y'],
      dtype='object')

In [21]:
# drop unnecessary columns for categorization step later
df_agg = df_agg.drop(['Serial Number'], axis = 1)
df_agg = df_agg.drop(['City'], axis = 1)
df_agg = df_agg.drop(['State'], axis = 1)
df_agg = df_agg.drop(['Zip'], axis = 1)
df_agg = df_agg.drop(['Inspection Time'], axis = 1)
df_agg = df_agg.drop(['Employee ID'], axis = 1)
df_agg = df_agg.drop(['Inspection Type'], axis = 1)
df_agg = df_agg.drop(['Location 1'], axis = 1)
df_agg = df_agg.drop(['Coordinates'], axis = 1)
df_agg = df_agg.drop(['Record Updated'], axis = 1)
df_agg = df_agg.drop(['Inspection Date'], axis = 1)

In [22]:
df_agg

Restaurant Name  \
0                                   #1 HAWAIIAN BARBEQUE   
1                                                $5 CAFE   
2                    100 DEGREES HOT POT - DELETED 10/15   
3                                     101 MART SNACK BAR   
4                                    168 Market - Bakery   
5                                168 Market - Restaurant   
6                        168 Market / Produce Department   
7                             17 SOUTH BOOZE & BITES BAR   
8                           17 SOUTH BOOZE & BITES GRILL   
9                                     1900 ASIAN CUISINE   
10                                      1923 BOURBON BAR   
11                       1923 BOURBON ISLAND BAR/SUPPORT   
12                                1923 BOURBON STAGE BAR   
13                                  1ST FLOOR BAR- BOMAN   
14                          1ST FLOOR EAST BAR- REYNOLDS   
15                            1ST FLOOR PANTRY- REYNOLDS   
16                          1ST FLOOR WEST BAR- REYNOLDS   
17                                 2 AMIGOS TACOS MOBILE   
18                                            215 Saloon   
19                           24 HOUR MINI MART SNACK BAR   
20                          24 Hour Fitness Smoothie Bar   
21                                             24 LOUNGE   
22                              24X7 MINI MART SNACK BAR   
23                                   28Go the Restaurant   
24                                  2ND FLOOR BAR- BOMAN   
25                      2ND FLOOR FOUNDERS BAR- REYNOLDS   
26                    2ND FLOOR FOUNDERS PANTRY REYNOLDS   
27                     2ND FLOOR MEZZ GRAB N GO-REYNOLDS   
28                    2ND FLOOR MEZZ NORTH BAR- REYNOLDS   
29                       2ND FLOOR MEZZ PANTRY- REYNOLDS   
...                                                  ...   
13860                               ZEFFIRINO RESTAURANT   
13861                            ZEFFIRINO SERVICE BAR 1   
13862                         ZENSHIN ASIAN PREP KITCHEN   
13863                           ZENSHIN ASIAN RESTAURANT   
13864                            ZENSHIN ASIAN SUSHI BAR   
13865     ZENSHIN ASIAN WAREWASH/BEV STATION/DRY STORAGE   
13866                            ZEST BISTRO & BAR - BAR   
13867                   ZEST BISTRO & BAR - PREP KITCHEN   
13868                     ZEST BISTRO & BAR - RESTAURANT   
13869                                 ZIKIZ COFFEE HOUSE   
13870                      ZINGERS BEST BURGERS ON EARTH   
13871                  ZINGERS BEST BURGERS ON EARTH BAR   
13872                             ZOMBIE DOGS @ HARD HAT   
13873  ZOMBIE DOGS PUSF - DELETED 7/29/16 APPLYING FO...   
13874                                          ZONA ROSA   
13875                                     ZYDECO PO-BOYS   
13876                 Zaytoon Mediterranean - Restaurant   
13877                                          Zizzy Bar   
13878                                   Zizzy Restaurant   
13879                                      Zodies Lounge   
13880              Zoological - Botanical Park Snack Bar   
13881                 at Your Service Catering Snack Bar   
13882                               in N Out Burger #104   
13883                               in N Out Burger #226   
13884                                in N Out Burger #86   
13885                                off the Strip - Bar   
13886                         off the Strip - Restaurant   
13887                        via Brasil Steakhouse - Bar   
13888                   via Brasil Steakhouse - Dishroom   
13889                 via Brasil Steakhouse - Restaurant   

                Permit Number  \
0                 [PR0110047]   
1                 [PR0017837]   
2                 [PR0113718]   
3                 [PR0118514]   
4                 [PR0021447]   
5                 [PR0021446]   
6                 [PR0021445]   
7                 [PR0109450]   
8                 [PR01

In [23]:
dfcategories = df_agg.columns
for x in dfcategories[1:]:
    for y in range(len(df_agg[x])):
        df_agg[x][y] = list(set(df_agg[x][y]))
        df_agg[x][y] = df_agg[x][y][0]

In [24]:
df_agg

Restaurant Name Permit Number  \
0                                   #1 HAWAIIAN BARBEQUE     PR0110047   
1                                                $5 CAFE     PR0017837   
2                    100 DEGREES HOT POT - DELETED 10/15     PR0113718   
3                                     101 MART SNACK BAR     PR0118514   
4                                    168 Market - Bakery     PR0021447   
5                                168 Market - Restaurant     PR0021446   
6                        168 Market / Produce Department     PR0021445   
7                             17 SOUTH BOOZE & BITES BAR     PR0109450   
8                           17 SOUTH BOOZE & BITES GRILL     PR0109449   
9                                     1900 ASIAN CUISINE     PR0099062   
10                                      1923 BOURBON BAR     PR0110263   
11                       1923 BOURBON ISLAND BAR/SUPPORT     PR0108949   
12                                1923 BOURBON STAGE BAR     PR0108950   
13                                  1ST FLOOR BAR- BOMAN     PR0021200   
14                          1ST FLOOR EAST BAR- REYNOLDS     PR0021207   
15                            1ST FLOOR PANTRY- REYNOLDS     PR0021208   
16                          1ST FLOOR WEST BAR- REYNOLDS     PR0021206   
17                                 2 AMIGOS TACOS MOBILE     PR0119469   
18                                            215 Saloon     PR0024098   
19                           24 HOUR MINI MART SNACK BAR     PR0106164   
20                          24 Hour Fitness Smoothie Bar     PR0012767   
21                                             24 LOUNGE     PR0098970   
22                              24X7 MINI MART SNACK BAR     PR0021102   
23                                   28Go the Restaurant     PR0024320   
24                                  2ND FLOOR BAR- BOMAN     PR0021201   
25                      2ND FLOOR FOUNDERS BAR- REYNOLDS     PR0021209   
26                    2ND FLOOR FOUNDERS PANTRY REYNOLDS     PR0021210   
27                     2ND FLOOR MEZZ GRAB N GO-REYNOLDS     PR0104678   
28                    2ND FLOOR MEZZ NORTH BAR- REYNOLDS     PR0021212   
29                       2ND FLOOR MEZZ PANTRY- REYNOLDS     PR0021213   
...                                                  ...           ...   
13860                               ZEFFIRINO RESTAURANT     PR0009057   
13861                            ZEFFIRINO SERVICE BAR 1     PR0009059   
13862                         ZENSHIN ASIAN PREP KITCHEN     PR0105401   
13863                           ZENSHIN ASIAN RESTAURANT     PR0098470   
13864                            ZENSHIN ASIAN SUSHI BAR     PR0098471   
13865     ZENSHIN ASIAN WAREWASH/BEV STATION/DRY STORAGE     PR0105403   
13866                            ZEST BISTRO & BAR - BAR     PR0017887   
13867                   ZEST BISTRO & BAR - PREP KITCHEN     PR0098325   
13868                     ZEST BISTRO & BAR - RESTAURANT     PR0017886   
13869                                 ZIKIZ COFFEE HOUSE     PR0120264   
13870                      ZINGERS BEST BURGERS ON EARTH     PR0014610   
13871                  ZINGERS BEST BURGERS ON EARTH BAR     PR0014611   
13872                             ZOMBIE DOGS @ HARD HAT     PR0118834   
13873  ZOMBIE DOGS PUSF - DELETED 7/29/16 APPLYING FO...     PR0109033   
13874                                          ZONA ROSA     PR0020519   
13875                                     ZYDECO PO-BOYS     PR0112724   
13876                 Zaytoon Mediterranean - Restaurant     PR0020288   
13877                                          Zizzy Bar     PR0015979   
13878                                   Zizzy Restaurant     PR0015978   
13879                                      Zodies Lounge     PR0007654   
13880              Zoological - Botanical Park Snack Bar     PR0007240   
13881                 at Your Service Catering Snack Bar     PR0005647   
13882                               in N Out 

In [25]:
try:
    with open('non_0_updated_sick_scores','rb') as f:
        non_0_updated_sick_scores = pickle.load(f)
        
except FileNotFoundError:
    with open('non_0_updated_sick_scores','wb') as f:
        pickle.dump(non_0_sick_scores, f)

In [26]:
file = 'yelp_academic_dataset_business.json'
business_locations = {}
lv_restaurant_ids = []
with open(file) as f:
    for line in f:
        r_line = json.loads(line)
        business_locations.update({r_line['business_id']:r_line['full_address']})
        if 'Restaurants' in r_line['categories']:
            lv_restaurant_ids.append(r_line['business_id'])

In [27]:
sick_ids = []
for x, y in non_0_updated_sick_scores.items():
    sick_ids.append(x)
len(sick_ids)

1942

In [28]:
addresses = []
for x, y in business_locations.items():
    if x in sick_ids:
        addresses.append(y.rsplit('\n', 5)[0].lower())
len(addresses)

1942

In [29]:
addresses

['south point hotel & casino',
 '1916 village center cir',
 'renaissance las vegas',
 '8410 w cheyenne ave',
 '10690 southern highlands pkwy',
 'palms casino resort',
 'mgm grand',
 '410 s rampart blvd',
 '2620 regatta dr',
 "harrah's las vegas hotel & casino",
 '1383 e silverado ranch blvd',
 '2345 e centennial pkwy',
 'fashion show mall',
 '5115 w spring mountain rd',
 '1031 s rampart blvd',
 'rio hotel and casino',
 '2101 n rainbow blvd',
 '1260 n town center dr',
 'fashion show mall',
 '6115 s fort apache rd',
 '7620 las vegas boulevard',
 '8975 s eastern ave',
 'planet hollywood resort & casino',
 'mandalay bay resort and casino',
 '2025 village center cir',
 'treasure island',
 'rio all-suite hotel & casino',
 '4670 s decatur blvd',
 'aria resort & casino',
 '6630 n durango dr',
 '9410 w sahara ave',
 '4029 spring mountain rd',
 '4355 spring mountain rd',
 'new york new york hotel & casino',
 '4711 spring mountain rd',
 '7777 w sahara ave',
 '3230 e flamingo rd',
 '520 fremont st

In [30]:
# create lists for addresses of businesses with violations in order to 
# check for overlap between the Las Vegas Violation set and the yelp set.

violation_addresses = []
overlap = []

for line in df_agg['Address']:
    violation_addresses.append(str(line).lower())
violation_addresses[:10]

for line in addresses:
    if line in violation_addresses:
        overlap.append(line)
print(addresses[:20])
print(overlap[:20])

['south point hotel & casino', '1916 village center cir', 'renaissance las vegas', '8410 w cheyenne ave', '10690 southern highlands pkwy', 'palms casino resort', 'mgm grand', '410 s rampart blvd', '2620 regatta dr', "harrah's las vegas hotel & casino", '1383 e silverado ranch blvd', '2345 e centennial pkwy', 'fashion show mall', '5115 w spring mountain rd', '1031 s rampart blvd', 'rio hotel and casino', '2101 n rainbow blvd', '1260 n town center dr', 'fashion show mall', '6115 s fort apache rd']
['8410 w cheyenne ave', '10690 southern highlands pkwy', '2620 regatta dr', '1383 e silverado ranch blvd', '1031 s rampart blvd', '2101 n rainbow blvd', '1260 n town center dr', '6115 s fort apache rd', '8975 s eastern ave', '2025 village center cir', '4670 s decatur blvd', '6630 n durango dr', '9410 w sahara ave', '4355 spring mountain rd', '4711 spring mountain rd', '7777 w sahara ave', '520 fremont st', '3055 e flamingo rd', '8540 w lake mead blvd', '7435 s eastern ave']


In [31]:
print(len(overlap))

911


It looks like there are a total of 911 unique restaurant addresses that have an evaluated sick score greater than 0 and entries in the inspection dataset.

In [32]:
categories = df_agg.columns
df_agg = df_agg.drop(['Inspection Grade'], axis=1)
df_agg = df_agg.drop(['Inspection Demerits'], axis=1)              

In [33]:
df_agg

Restaurant Name Permit Number  \
0                                   #1 HAWAIIAN BARBEQUE     PR0110047   
1                                                $5 CAFE     PR0017837   
2                    100 DEGREES HOT POT - DELETED 10/15     PR0113718   
3                                     101 MART SNACK BAR     PR0118514   
4                                    168 Market - Bakery     PR0021447   
5                                168 Market - Restaurant     PR0021446   
6                        168 Market / Produce Department     PR0021445   
7                             17 SOUTH BOOZE & BITES BAR     PR0109450   
8                           17 SOUTH BOOZE & BITES GRILL     PR0109449   
9                                     1900 ASIAN CUISINE     PR0099062   
10                                      1923 BOURBON BAR     PR0110263   
11                       1923 BOURBON ISLAND BAR/SUPPORT     PR0108949   
12                                1923 BOURBON STAGE BAR     PR0108950   
13                                  1ST FLOOR BAR- BOMAN     PR0021200   
14                          1ST FLOOR EAST BAR- REYNOLDS     PR0021207   
15                            1ST FLOOR PANTRY- REYNOLDS     PR0021208   
16                          1ST FLOOR WEST BAR- REYNOLDS     PR0021206   
17                                 2 AMIGOS TACOS MOBILE     PR0119469   
18                                            215 Saloon     PR0024098   
19                           24 HOUR MINI MART SNACK BAR     PR0106164   
20                          24 Hour Fitness Smoothie Bar     PR0012767   
21                                             24 LOUNGE     PR0098970   
22                              24X7 MINI MART SNACK BAR     PR0021102   
23                                   28Go the Restaurant     PR0024320   
24                                  2ND FLOOR BAR- BOMAN     PR0021201   
25                      2ND FLOOR FOUNDERS BAR- REYNOLDS     PR0021209   
26                    2ND FLOOR FOUNDERS PANTRY REYNOLDS     PR0021210   
27                     2ND FLOOR MEZZ GRAB N GO-REYNOLDS     PR0104678   
28                    2ND FLOOR MEZZ NORTH BAR- REYNOLDS     PR0021212   
29                       2ND FLOOR MEZZ PANTRY- REYNOLDS     PR0021213   
...                                                  ...           ...   
13860                               ZEFFIRINO RESTAURANT     PR0009057   
13861                            ZEFFIRINO SERVICE BAR 1     PR0009059   
13862                         ZENSHIN ASIAN PREP KITCHEN     PR0105401   
13863                           ZENSHIN ASIAN RESTAURANT     PR0098470   
13864                            ZENSHIN ASIAN SUSHI BAR     PR0098471   
13865     ZENSHIN ASIAN WAREWASH/BEV STATION/DRY STORAGE     PR0105403   
13866                            ZEST BISTRO & BAR - BAR     PR0017887   
13867                   ZEST BISTRO & BAR - PREP KITCHEN     PR0098325   
13868                     ZEST BISTRO & BAR - RESTAURANT     PR0017886   
13869                                 ZIKIZ COFFEE HOUSE     PR0120264   
13870                      ZINGERS BEST BURGERS ON EARTH     PR0014610   
13871                  ZINGERS BEST BURGERS ON EARTH BAR     PR0014611   
13872                             ZOMBIE DOGS @ HARD HAT     PR0118834   
13873  ZOMBIE DOGS PUSF - DELETED 7/29/16 APPLYING FO...     PR0109033   
13874                                          ZONA ROSA     PR0020519   
13875                                     ZYDECO PO-BOYS     PR0112724   
13876                 Zaytoon Mediterranean - Restaurant     PR0020288   
13877                                          Zizzy Bar     PR0015979   
13878                                   Zizzy Restaurant     PR0015978   
13879                                      Zodies Lounge     PR0007654   
13880              Zoological - Botanical Park Snack Bar     PR0007240   
13881                 at Your Service Catering Snack Bar     PR0005647   
13882                               in N Out 

In [58]:
df_cat = pd.get_dummies(df_agg['Category Name'], prefix = 'category')
df_grade = pd.get_dummies(df_agg['Current Grade'], prefix = 'grade_cur')
df_perm = pd.get_dummies(df_agg['Permit Status'], prefix = 'permit')
df_res = pd.get_dummies(df_agg['Inspection Result'], prefix = 'result')
df_dems = pd.get_dummies(df_agg['Current Demerits'], prefix = 'demerits')
df_new = pd.concat([df_agg, df_cat], axis=1)
df_new = pd.concat([df_new, df_grade], axis=1)
df_new = pd.concat([df_new, df_perm], axis=1)
df_new = pd.concat([df_new, df_res], axis=1)
df_new = pd.concat([df_new, df_dems], axis=1)
df_new = df_new.drop(['Category Name'], axis = 1 )
df_new = df_new.drop(['Current Grade'], axis = 1 )
df_new = df_new.drop(['Permit Status'], axis = 1 )
df_new = df_new.drop(['Inspection Result'], axis = 1)
df_new = df_new.drop(['Date Current'], axis = 1)
df_new = df_new.drop(['Restaurant Name'], axis = 1)
df_new = df_new.drop(['Permit Number'], axis = 1)
df_new = df_new.drop(['Location Name'], axis = 1)
df_new = df_new.drop(['Address'], axis = 1)
df_new = df_new.drop(['Violations'], axis = 1)

In [59]:
df_new

Current Demerits        X        Y  category_Bakery Sales  \
0                    9   36.082  -115.12                    0.0   
1                   10  36.1459 -115.156                    0.0   
2                    3  36.1265 -115.218                    0.0   
3                    6  36.1322 -115.155                    0.0   
4                    3  36.1266 -115.225                    1.0   
5                    0  36.1266 -115.225                    0.0   
6                    6  36.1266 -115.225                    0.0   
7                    0  36.0597 -115.172                    0.0   
8                    9  36.0597 -115.172                    0.0   
9                    0  36.1255 -115.211                    0.0   
10                   0  36.0913 -115.174                    0.0   
11                   9  36.0913 -115.174                    0.0   
12                   0  36.0913 -115.174                    0.0   
13                   3  36.1684 -115.153                    0.0   
14                   3  36.1684 -115.153                    0.0   
15                   0  36.1684 -115.153                    0.0   
16                   8  36.1684 -115.153                    0.0   
17                   0  36.1756 -115.116                    0.0   
18                   9  36.1148 -115.307                    0.0   
19                   6    36.13 -115.198                    0.0   
20                   3  36.1895 -115.305                    0.0   
21                   6  36.1149 -115.207                    0.0   
22                   0  36.1595 -115.195                    0.0   
23                   3   36.105 -115.136                    0.0   
24                   3  36.1684 -115.153                    0.0   
25                   8  36.1684 -115.153                    0.0   
26                   3  36.1684 -115.153                    0.0   
27                   0  36.1684 -115.153                    0.0   
28                   6  36.1684 -115.153                    0.0   
29                   8  36.1684 -115.153                    0.0   
...                ...      ...      ...                    ...   
13860                0  36.1221 -115.168                    0.0   
13861                0  36.1221 -115.168                    0.0   
13862                3  36.0115 -115.175                    0.0   
13863                3  36.0115 -115.175                    0.0   
13864                0  36.0115 -115.175                    0.0   
13865                0  36.0115 -115.175                    0.0   
13866                3  35.9951 -115.207                    0.0   
13867                6  35.9951 -115.207                    0.0   
13868               17  35.9951 -115.207                    0.0   
13869                5  36.0208 -115.277                    0.0   
13870                3  36.1065 -115.172                    0.0   
13871                3  36.1065 -115.172                    0.0   
13872                0  36.1511  -115.16                    0.0   
13873                3  36.1389 -115.171                    0.0   
13874               10  36.1212  -115.15                    0.0   
13875                6  36.1674 -115.139                    0.0   
13876                3  36.1234 -115.279                    0.0   
13877                3  36.1259 -115.198                    0.0   
13878                0  36.1259 -115.198                    0.0   
13879                0  36.1147 -115.108                    0.0   
13880                9  36.1922  -115.19                    0.0   
13881                0  36.2248 -115.281                    0.0   
13882                0  36.1956 -115.247                    0.0   
13883                0  36.1007 -115.302                    0.0   
13884                0  36.1013 -115.182                    0.0   
13885                3     36.1 -115.307                    0.0   
13886                8     36.1 -115.307                    0.0   
13887                0  36.1571 -115.292             

In [60]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.cross_validation import train_test_split

X = df_new.loc[:,:].values
y = df_new['grade_cur_X'].values
X_train, X_test, y_train, y_test = train_test_split(X, y)
knn = KNeighborsClassifier(n_neighbors=4)
knn.fit(X_train,y_train)
pred = knn.predict(X_test)
print(pred)
test_score = knn.score(X_test,y_test)
print('Accuracy on test data: {}'.format(test_score))


[ 0.  0.  0. ...,  0.  0.  0.]
Accuracy on test data: 0.9997120644975526


In [62]:
X = df_new.loc[:,:].values
y = df_new['grade_cur_A'].values
X_train, X_test, y_train, y_test = train_test_split(X, y)
knn = KNeighborsClassifier(n_neighbors=4)
knn.fit(X_train,y_train)
pred = knn.predict(X_test)
print(pred)
test_score = knn.score(X_test,y_test)
print('Accuracy on test data: {}'.format(test_score))

[ 1.  1.  1. ...,  1.  1.  1.]
Accuracy on test data: 0.9979844514828679
